apps/admin-ui/Dockerfile

```diff
@@ -0,0 +1,42 @@
+FROM node:18-bullseye AS base
+
+FROM base AS builder
+WORKDIR /app
+RUN npm install -g turbo
+COPY . .
+RUN turbo prune --scope=admin-ui --docker
+
+# Add lockfile and package.json's of isolated subworkspace
+FROM base AS installer
+WORKDIR /app
+# First install the dependencies (as they change less often)
```

Comment:
If I understand this correctly, this comment indicates that the below `pnpm install --frozen-lockfile` will benefit from docker layer caching, since if only the app changes then we don't need to rebuild dependencies?

This is not the case currently, as on line 6 we do `COPY . .`, which will copy all files in the project to the `builder`, invalidating the layer cache at that point. Then on line 14 and 15 we copy things from the builder, which will invalidates `installer`'s layer cache at that point, triggering a rebuild of dependencies.

You can test this by changing something in `apps/admin-ui/src/App.tsx` for instance. If you want better layer caching, I think you need to separate the building of dependencies to its own stage, or somehow copy less files, or at a later time.

Reply:
This comment seems old, so closing as resolved. Will look through the newer Dockerfile again.